services/auth-service/src/main/java/com/sourcefuse/jarc/services/authservice/oauth2/providers/OAuth2PreVerifyProvider.java

```diff
@@ -0,0 +1,92 @@
+package com.sourcefuse.jarc.services.authservice.oauth2.providers;
+
+import com.sourcefuse.jarc.core.enums.UserStatus;
+import com.sourcefuse.jarc.services.authservice.enums.AuthErrorKeys;
+import com.sourcefuse.jarc.services.authservice.models.User;
+import com.sourcefuse.jarc.services.authservice.models.UserTenant;
+import com.sourcefuse.jarc.services.authservice.oauth2.user.OAuth2UserInfo;
+import com.sourcefuse.jarc.services.authservice.repositories.UserRepository;
+import com.sourcefuse.jarc.services.authservice.repositories.UserTenantRepository;
+import com.sourcefuse.jarc.services.authservice.specifications.UserTenantSpecification;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.oauth2.core.oidc.user.OidcUser;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpServerErrorException;
+
+@AllArgsConstructor
+@Service
+public class OAuth2PreVerifyProvider {
+
+  private final UserRepository userRepository;
+  private final UserTenantRepository userTenantRepository;
+
+  public User provide(User user, OidcUser oidcUser) {
+    if (
+      !user.getFirstName().equals(oidcUser.getGivenName()) ||
+      !user.getLastName().equals(oidcUser.getFamilyName()) ||
+      !user.getUsername().equals(oidcUser.getPreferredUsername()) ||
+      !user.getEmail().equals(oidcUser.getEmail())
```

Comment:
since user will not save null in his name , email and username in google, github or any oauth2 provider vendor's profile it won't cause the issue , but in case sensitive if user is saving his name by changing cases in any of the oauth2 vendor's profile it will be better if we catch those changes and save in our db